src/main/java/me/jellysquid/mods/lithium/common/util/collections/HashedList.java

```diff
@@ -16,6 +16,11 @@ private HashedList(List<T> list, Set<T> set) {
         this.set = set;
     }
 
+    public HashedList(List<T> list) {
+        this.list = list;
+        this.set = new HashSet<>();
```

Comment:
this.set.addAll(list) is missing here. You are expecting the list to be empty, but others who use the function in the future might not expect that.